### ISSUE TITLE
E2E test apm.TestAPMKibanaAssociation fails in 7.13.0-SNAPSHOT

### ISSUE THREAD

**Issue by thbkrkr** [elastic/cloud-on-k8s#4390]
https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests-snapshot-versions/283/

TestAPMKibanaAssociation fails because ES cluster health is yellow because there are 7 unassigned shards (`.fleet-*`).

```go
=== RUN   TestAPMKibanaAssociation/ES_cluster_health_should_eventually_be_green
Retries (30m0s timeout):
    step.go:43: 
        	Error Trace:	utils.go:86
        	Error:      	Received unexpected error:
        	            	health is yellow
        	Test:       	TestAPMKibanaAssociation/ES_cluster_health_should_eventually_be_green
{"log.level":"error","@timestamp":"2021-03-30T02:02:11.210Z","message":"stopping early","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:44\ngithub.com/elastic/cloud-on-k8s/test/e2e/apm.TestAPMKibanaAssociation\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/apm/association_test.go:82\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1194"}
    --- FAIL: TestAPMKibanaAssociation/ES_cluster_health_should_eventually_be_green (1800.00s)
```

```json
# cluster_health.json
{
  "cluster_name" : "test-apm-kb-assoc-lrww",
  "status" : "yellow",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "active_primary_shards" : 16,
  "active_shards" : 16,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 7,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 69.56521739130434
}
```

```sh
> grep UNASSIGNED cat/cat_shards.txt 
.fleet-actions_1                         0     r      UNASSIGNED                          
.fleet-agents_1                          0     r      UNASSIGNED                          
.fleet-artifacts_1                       0     r      UNASSIGNED                          
.fleet-enrollment-api-keys_1             0     r      UNASSIGNED                          
.fleet-policies-leader_1                 0     r      UNASSIGNED                          
.fleet-policies_1                        0     r      UNASSIGNED                          
.fleet-servers_1                         0     r      UNASSIGNED 
```

To reproduce:
```
E2E_STACK_VERSION=8.0.0-SNAPSHOT TESTS_MATCH=TestAPMKibanaAssociation m e2e-local
```

**Comment by pebrc** [comment]
I think the problem here is that the new .fleet* indices are not created with `auto_expand_replicas:0-1` but with a fixed replica which leads to `yellow` cluster health from the get go for single node clusters. I see if we can open an issue against Beats.

**Comment by thbkrkr** [comment]
Closed by elastic/kibana#95892 :tada:
